patches/server/0727-Add-advancement-display-API.patch

```diff
@@ -0,0 +1,128 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: syldium <syldium@mailo.com>
+Date: Fri, 9 Jul 2021 18:50:40 +0200
+Subject: [PATCH] Add advancement display API
+
+
+diff --git a/src/main/java/io/papermc/paper/advancement/PaperAdvancementDisplay.java b/src/main/java/io/papermc/paper/advancement/PaperAdvancementDisplay.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..333adb79547dd1c17887b7063150ff921587dc61
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/advancement/PaperAdvancementDisplay.java
+@@ -0,0 +1,50 @@
++package io.papermc.paper.advancement;
++
++import io.papermc.paper.adventure.PaperAdventure;
++import net.kyori.adventure.text.Component;
++import net.minecraft.advancements.DisplayInfo;
++import org.bukkit.craftbukkit.inventory.CraftItemStack;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++
++public record PaperAdvancementDisplay(DisplayInfo handle) implements AdvancementDisplay {
++
++    @Override
++    public @NotNull Frame frame() {
++        return switch (this.handle.getFrame()) {
++            case TASK -> Frame.TASK;
++            case CHALLENGE -> Frame.CHALLENGE;
++            case GOAL -> Frame.GOAL;
++        };
++    }
++
++    @Override
++    public @NotNull Component title() {
++        return PaperAdventure.asAdventure(this.handle.getTitle());
```

Comment:
Perhaps both this and the description should be cached? I’m unsure so this isn’t a needed change for the moment.